dluhc-component-library/src/components/timetablePage/TimetablePage.tsx

```diff
@@ -1,20 +1,33 @@
 import { useEffect, useState } from "preact/hooks";
 import Timetable from "./Timetable";
+import csvToJson from "csvtojson";
 
 interface TimetablePageProps {
   filepath: string;
 }
 
-const loadData = async (filepath: string) =>
+
+const loadJson = async (filepath: string) =>
   await fetch(filepath).then((res) => res.json());
 
-const TimetablePage = ({ filepath }: TimetablePageProps) => {
+const loadCSV = async (filepath: string) =>
+  await fetch(filepath).then((res) => res.text());
+
+const TimetablePage = ( { filepath }: TimetablePageProps) => {
   const [timetableData, setTimetableData] = useState();
 
   useEffect(() => {
-    loadData(filepath).then((data) => {
-      setTimetableData(data);
-    });
+    if (/.csv$/.test(filepath)) {
+      loadCSV(filepath).then((data) => {
+        csvToJson()
+          .fromString(data)
+          .then((jsonObj: any) => setTimetableData(jsonObj));
```

Comment:
Can we type this, or maybe even just remove "any" and the compiler might be able to work out the type for you.